README.md

```diff
@@ -138,20 +139,32 @@ But first, we need to subscribe:
 CanardRxSubscription heartbeat_subscription;
 (void) canardRxSubscribe(&ins,   // Subscribe to messages uavcan.node.Heartbeat.
                          CanardTransferKindMessage,
-                         32085,  // The fixed Subject-ID of the Heartbeat message type (see DSDL definition).
-                         7,      // The maximum payload size (max DSDL object size) from the DSDL definition.
+                         7509,   // The fixed Subject-ID of the Heartbeat message type (see DSDL definition).
+                         16,     // The extent (the maximum possible payload size); pick a huge value if not sure.
                          CANARD_DEFAULT_TRANSFER_ID_TIMEOUT_USEC,
                          &heartbeat_subscription);
 
 CanardRxSubscription my_service_subscription;
-(void) canardRxSubscribe(&ins,                        // Subscribe to an arbitrary service response.
-                         CanardTransferKindResponse,
-                         123,                         // The Service-ID to subscribe to.
-                         1024,                        // The maximum payload size (max DSDL object size).
+(void) canardRxSubscribe(&ins,   // Subscribe to an arbitrary service response.
+                         CanardTransferKindResponse,  // Specify that we want service responses, not requests.
+                         123,    // The Service-ID whose responses we will receive.
+                         1024,   // The extent (the maximum payload size); pick a huge value if not sure.
                          CANARD_DEFAULT_TRANSFER_ID_TIMEOUT_USEC,
                          &my_service_subscription);
 ```
 
+The "extent" refers to the minimum amount of memory required to hold any serialized representation of any compatible
```

Comment:
Extend is a bit ambiguous, in the code extend refers to the maximum payload size, whereas this line refers to the minimum amount of memory required. Furthermore I think it's also interesting to know what the implications are when extend is too small and when you put a very large value into the extend variable with this have performance/overflow implications?

Reply:
I just pushed a couple of extra comments to clarify this.

> in the code extend refers to the maximum payload size, whereas this line refers to the minimum amount of memory required

"maximum payload size" and "minimum amount of memory required" are the same thing.

> I think it's also interesting to know what the implications are when extend is too small

Excess data is silently truncated away (see Specification "implicit truncation rule")

> when you put a very large value into the extend variable will this have performance [...] implications?

The docs say that the extent defines the amount of memory allocated to store received objects:

> 2. New memory for the transfer payload buffer is allocated when a new transfer is initiated, unless the buffer was already allocated at the time. This event occurs when a transport frame that matches a known subscription is received and it begins a new transfer (that is, the start-of-frame flag is set and it is not a duplicate). The amount of the allocated memory equals the extent as configured via canardRxSubscribe(); please read its documentation for further information about the extent and related edge cases.

Larger extent --> more memory needed.

This is different from conventional non-real-time serialization formats where the memory is allocated and reallocated ad-hoc depending on the size of the received object. Such behavior does not require tinkering with extent but it may not be compatible with real-time constraints.